esx_service/utils/auth_data.py

```diff
@@ -0,0 +1,306 @@
+                (name, tenant_id)
+        )
+        conn.commit()
+    
+    def set_description(self, conn, description):
+        tenant_id = self.id
+        conn.execute(
+                "update tenants set description=? where id=?", 
+                (description, tenant_id)
+            )
+        conn.commit()
+        
+    def set_default_datastore_and_privileges(self, conn, datastore, privileges):
+        tenant_id = self.id
+        exist_default_datastore = self.default_datastore
+        conn.execute(
+                "update tenants set default_datastore=? where id=?", 
+                (datastore, tenant_id)
+        )
+
+        # remove the old entry
+        conn.execute(
+                "delete from privileges where tenant_id=? and datastore=?", 
+                [tenant_id, exist_default_datastore]
+        )
+
+        for p in privileges:
+            p['tenant_id'] = tenant_id
+        conn.executemany(
+            """
+            insert into privileges values
+            (:tenant_id, :datastore, :global_visibility, :create_volume,
+            :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            privileges
+        )
+
+        conn.commit()
+            
+    def set_datastore_access_privileges(self, conn, privileges):
+        # If we want to change privileges for <tenant1, ds1> and <tenant2, ds2>
+        # Do we call this function one time or two times?
+        tenant_id = self.id
+        for p in privileges:
+            p['tenant_id'] = tenant_id
+                            
+        conn.executemany(
+            """
+            insert or ignore into privileges values
+            (:tenant_id, :datastore, :global_visibility, :create_volume,
+            :delete_volume, :mount_volume, :max_volume_size, :usage_quota)
+            """,
+            privileges
+        )
+        
+        for p in privileges:
+            p['tenant_id'] = tenant_id
+            column_list = ['tenant_id', 'datastore', 'global_visibility', 'create_volume',
+                            'delete_volume', 'mount_volume', 'max_volume_size', 'usage_quota']
+            update_list = []
+            for col in column_list:
+                update_list.append(p[col])
+            update_list.append(tenant_id)
+            update_list.append(p['datastore'])    
+            
+            #print update_list
+
+            conn.execute(
+                """
+                update or ignore privileges set
+                tenant_id =?, 
+                datastore=?, 
+                global_visibility=?,
+                create_volume=?,
+                delete_volume=?,
+                mount_volume=?,
+                max_volume_size=?,
+                usage_quota=?
+                where tenant_id=? and datastore=?
+                """,
+                update_list
+            )
+        conn.commit()
+
+class AuthorizationDataManager:
+    """
```

Comment:
Nice - docsting is present ! Would be nice to add init args. 
Also - do you mean it abstracts the actual persistence of the auth data or something else ?